worker/containerd/client.go

```diff
@@ -1,4 +1,4 @@
-package libcontainerd
+package containerd
```

Comment:
Had not considered the auto-import problem for editors. 

We were initially not sure if `libcontainerd` was an the actual `containerd` client or a wrapper around the `containerd` client. But after reading the following line about the [adapter pattern](https://en.wikipedia.org/wiki/Adapter_pattern) I'm realizing this was okay.

> Clients don't know whether they work with a target class directly or through an adapter with a class that does not have the target interface. 

So maybe we should just put the client back in its own package, and call it `libcontainerd` again. The directory called `containerd` (or whatever it's renamed to pending conversation below) could then just be a directory, and not have a corresponding package with that name.

Reply:
> especially if you ever have to deal with both

This is a confusion I have about employing the adapter pattern. Is it normal to reference both the adapter and the target it's adapting in the same code?

We noticed there's a single [places](https://github.com/concourse/concourse/blob/3babedbeab761fc601831197f268e04a1e85df22/worker/backend/backend.go#L165) in the Garden backend implementation where we're having to use the original `containerd` package directly alongside the adapter (and the package collision does happen there).

The adapter is specifically for the `Client` object in the `containerd` package, and the direct usage is calling an `options` function in the same package to pass into a constructor. So is it okay because those are two different things. @cirocosta any thoughts on this?

Reply:
I've been looking around for clear examples of package naming when it comes to the adapters but haven't been too successful.

One example I found was in Google's [cAdvisor](url), and the way they use [containerd](https://github.com/google/cadvisor/blob/9db8c7dee20a0c41627b208977ab192a0411bf93/container/containerd/client.go#L15). In the example they explicitly call their package to deal with containerd the same name as the target package.

The Kubelet code in `kubernetes/kubernetes` which in turn uses `cAdvisor` seems to do the [same](https://github.com/kubernetes/kubernetes/blob/d8a513ef999ee4cbdee66a5c9062cddce13584bb/pkg/kubelet/cadvisor/cadvisor_linux.go#L19).